File: src/main/java/com/chimbori/crux/common/Log.java
package com.chimbori.crux.common;

import org.jsoup.nodes.Node;

@SuppressWarnings("WeakerAccess")
public class Log {
  private static final boolean DEBUG = false;

  private static final boolean TRUNCATE = true;

  public static void i(String message, Object... args) {
    if (DEBUG) {
      System.err.println(String.format(message, args));
    }
  }

  public static void i(String reason, Node node) {
    if (DEBUG) {
      String nodeToString = TRUNCATE
          ? node.outerHtml().substring(0, Math.min(node.outerHtml().length(), 80)).replace("\n", "")
          : "\n------\n" + node.outerHtml() + "\n------\n";
      i("%s [%s]", reason, nodeToString);
    }
  }

  public static void printAndRemove(Node node, String reason) {
    i(reason, node);
    node.remove();
  }
}

File: crux-kmp/src/commonMain/kotlin/com/chimbori/crux_kmp/common/Log.kt
package com.chimbori.crux_kmp.common

import com.fleeksoft.ksoup.nodes.Node


internal object Log {
  private const val DEBUG = false

  private const val TRUNCATE = true

  fun i(message: String, vararg args: Any?) {
    if (DEBUG) {
      System.err.println(String.format(message, *args))
    }
  }

  fun i(reason: String, node: Node) {
    if (DEBUG) {
      val nodeToString = if (TRUNCATE) {
        node.outerHtml().take(80).replace("\n", "")
      } else {
        "\n------\n${node.outerHtml()}\n------\n"
      }
      i("%s [%s]", reason, nodeToString)
    }
  }

  fun printAndRemove(reason: String, node: Node) {
    i(reason, node)
    node.remove()
  }
}
